### PR TITLE
Feat: Event Page

### LINKED ISSUE #140: Feat: Event Page
**UPDATED:**
Implement the event page (see screenshots below)

We should show the event page whenever a user clicks "Learn More" on the schedule view or if a user clicks on an event from the landing screen(this is not yet implemented).

You should be able to enter from here:

Image

Please display all of the event information displayed here except for the "About" card (this is being implemented in another ticket).

Image

this should be for a general event not an occurrence

When the reserve button is pressed it should navigate back to the schedule view with a filter for that class applied:

Image

Image

### PR DESCRIPTION
**Pull request by vindema**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/140)
Implements the Event Page (/event/[id]) which displays full event details when a user taps "Learn More" on the map view's event card.     

# How Has This Been Tested?
- Opened the map/dashboard view and tapped a location pin to surface the event card                                                                   
- Tapped "Learn More" — confirmed navigation to /event/[id]
- Verified event image, title, location, category, and bookings count render correctly                                                                
- Confirmed reviews card shows aggregate rating + star display + first review text, category tags, and relative timestamp                                                                                          

# Checklist

## General
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [x] I have written unit tests for my code
- [x] I have written integration tests where appropriate
- [x] New and existing unit tests pass locally with my changes
- [x] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [x] I have tested error handling and edge cases
- [x] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [x] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [x] Mobile responsive design verified (if applicable)
- [x] I have run make generate-api to keep the API hooks up to date
- [x] **For mobile features:** Tested on both iOS and Android devices
- [x] **For mobile features:** Screenshots from both iOS and Android included below
- [x] No console errors or warnings
- [x] Loading states and error states handled

### iOS
Screenshot 2026-04-14 at 11 49 49 PM

### Android
Screenshot 2026-04-14 at 11 56 09 PM

### CODE REVIEW

**Review by adescoteaux1 — COMMENTED** [review]
Ok so slight issue, sorry the ticket was wrong. This page (the one that you get to from the Map View) should be the org page, and the even page would be navigated to from the schedule view. The cards that are shown on the map should also be orgs not events. This is kinda a big fix though sorry, lmk if you have questions. You could handle this flow for featurethon if you want

**Comment by josh-torre** [comment]
@vindema please see requested changes and please be aware that another PR that just got merged will have some changes you will need to integrate with. See PR #145 for the "about section"

**Comment by adescoteaux1** [comment]
Is it possible to change the routing of this page to be the Event ID instead of the Event Occurence ID?

**Comment by vindema** [comment]
> Is it possible to change the routing of this page to be the Event ID instead of the Event Occurence ID?

This is possible! Will go about implementing this right now!

**Comment by adescoteaux1** [comment]
Ratings dont seem to be loading on events page:
here is an example that does have reviews but none show when you click into the page
image
image
image

File: frontend/apps/mobile/package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "expo-web-browser": "~15.0.10",
     "i18next": "^25.8.18",
     "nativewind": "^4.2.1",
+    "react": "19.1.0",
```
Comment (adescoteaux1) [inline frontend/apps/mobile/package.json]: remove this, react is at a higher level

File: frontend/bun.lock [hunk 2]
```diff
@@ -44,6 +44,7 @@
         "expo-web-browser": "~15.0.10",
         "i18next": "^25.8.18",
         "nativewind": "^4.2.1",
+        "react": "19.1.0",
```
Comment (adescoteaux1) [inline frontend/bun.lock]: remove this, react is at a higher level

File: frontend/apps/mobile/app/(app)/event/[id].tsx [hunk 3]
```diff
@@ -1,295 +1,275 @@
...
 import { BookmarkButton } from "@/components/BookmarkButton";
-import { formatDuration } from "@/utils/format";
 import { useTranslation } from "react-i18next";
-import { ListItem } from "@/components/ListItem";
-import { AboutPage } from "@/components/AboutPage";
-import { useRouteInfo } from "expo-router/build/hooks";
+import MaterialIcons from "@expo/vector-icons/MaterialIcons";
 
-function formatAddress(occurrence: EventOccurrence) {
+function formatLocation(occurrence: EventOccurrence) {
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/event/[id].tsx]: i think this should be moved to a utils

**Review by adescoteaux1 — COMMENTED** [review]
some fixes:
- change the text at the top of the event details page to say Class instead of Classes
- if you click reserve (go to the schedule) and then click the back button it brings you to feed. It should route back to the event details page

**Review by adescoteaux1 — APPROVED** [review]
lgtm 🚀
